File: Android/_Viper Activity_/Code/controllers/VIPERController.java
//
//  __MODULENAME__Controller.java
//  __APPNAME__
//
//  Created by __USERNAME__ on __DATETIME__.
//  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
//

package NAMESPACE.modules._VIPER_.controllers;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

import java.util.List;

import NAMESPACE.database.DBHelper;
import NAMESPACE.modules._VIPER_.dao.VIPERDAO;
import NAMESPACE.modules._VIPER_.entity.VIPER;

public class VIPERController {

    public static void inserir(Context context, VIPER entity) {
        DBHelper helper = new DBHelper(context);
        try (SQLiteDatabase database = helper.getWritableDatabase()) {
            try {
                VIPERDAO dao = new VIPERDAO(database);
                dao.insert(entity);
            } finally {
                database.close();
            }
            helper.close();
        }
    }

    public static void alterar(Context context, VIPER entity) {
        DBHelper helper = new DBHelper(context);
        try (SQLiteDatabase database = helper.getWritableDatabase()) {
            try {
                VIPERDAO dao = new VIPERDAO(database);
                dao.update(entity);
            } finally {
                database.close();
            }
            helper.close();
        }
    }

    public static void excluir(Context context, int id) {
        DBHelper helper = new DBHelper(context);
        try (SQLiteDatabase database = helper.getWritableDatabase()) {
            try {
                VIPERDAO dao = new VIPERDAO(database);
                dao.delete(id);
            } finally {
                database.close();
            }
            helper.close();
        }
    }

    public static void excluirTodos(Context context) {
        DBHelper helper = new DBHelper(context);
        try (SQLiteDatabase database = helper.getWritableDatabase()) {
            try {
                VIPERDAO dao = new VIPERDAO(database);
                dao.deleteAll();
            } finally {
                database.close();
            }
            helper.close();
        }
    }

    public static List<VIPER> selecionarTodos(Context context) {
        try (DBHelper helper = new DBHelper(context);
             SQLiteDatabase database = helper.getReadableDatabase()) {
            try {
                VIPERDAO dao = new VIPERDAO(database);
                return dao.getAll();
            } finally {
                database.close();
            }
        }
    }

    public static VIPER selecionar(Context context, int id) {
        try (DBHelper helper = new DBHelper(context);
             SQLiteDatabase database = helper.getReadableDatabase()) {
            try {
                VIPERDAO dao = new VIPERDAO(database);
                return dao.getById(id);
            } finally {
                database.close();
            }
        }
    }

    public static void excluirTabela(Context context) {
        DBHelper helper = new DBHelper(context);
        try (SQLiteDatabase database = helper.getWritableDatabase()) {
            try {
                VIPERDAO dao = new VIPERDAO(database);
                dao.dropTable();
            } finally {
                database.close();
            }
            helper.close();
        }
    }
}

File: CSharp/_DDD .NET Core Default_/Code/webapi/VIPERController.cs
//
//  __MODULENAME__Controller.cs
//  __APPNAME__
//
//  Created by __USERNAME__ on __DATETIME__.
//  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
//

using NAMESPACE.Database;
using NAMESPACE.Entity;
using NAMESPACE.Repository;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using System.Collections.Generic;

namespace NAMESPACE.Controller.Controllers
{
    [Authorize]
    [Route("api/VIPER")]
    [ApiController]
    public class VIPERController : ControllerBase
    {
        private string _mensagem = "";

        [HttpPost]
        public IActionResult Post([FromBody] VIPER entity)
        {
            using (var context = new _NAMESPACE_Context())
            {
                using (var transacao = context.Database.BeginTransaction())
                {
                    using (var repository = new VIPERRepository(context))
                    {
						_mensagem = repository.Incluir(entity);
                        
                        if (_mensagem == "")
                            transacao.Commit();
                        else
                            transacao.Rollback();
                    }
                }
            }
            if (_mensagem != "")
            {
                return BadRequest(_mensagem);
            }
            return Ok();
        }
		
		[HttpPut]
        public IActionResult Put([FromBody] VIPER entity)
        {
            using (var context = new _NAMESPACE_Context())
            {
                using (var transacao = context.Database.BeginTransaction())
                {
                    using (var repository = new VIPERRepository(context))
                    {
                        _mensagem = repository.Alterar(entity);

                        if (_mensagem == "")
                            transacao.Commit();
                        else
                            transacao.Rollback();
                    }
                }
            }
            if (_mensagem != "")
            {
                return BadRequest(_mensagem);
            }
            return Ok();
        }

        [HttpDelete("{id}")]
        public IActionResult Delete(int id)
        {
            using (var context = new _NAMESPACE_Context())
            {
                using (var transacao = context.Database.BeginTransaction())
                {
                    using (var repository = new VIPERRepository(context))
                    {
                        _mensagem = repository.Excluir(id);
                        if (_mensagem == "")
                            transacao.Commit();
                        else
                            transacao.Rollback();
                    }
                }
            }
            if (_mensagem != "")
            {
                return BadRequest(_mensagem);
            }
            return Ok(); ;
        }

        [HttpGet("{id}")]
        public ActionResult<VIPER> Get(int id)
        {
            using var repository = new VIPERRepository();
            return repository.Selecionar(id);
        }

        [HttpGet]
        public IEnumerable<VIPER> GetAll()
        {
            using var repository = new VIPERRepository();
            return repository.SelecionarTodos();
        }

        [HttpGet("filter")]
        public IEnumerable<VIPER> Filter(string condicao)
        {
            using var repository = new VIPERRepository();
            return repository.Filtrar(condicao);
        }
    }
}
